### PR TITLE
chore(docs): add environment variables for docsearch

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

- Add environment variables to store Docsearch API Keys
- It's a blocker for https://github.com/aws-amplify/amplify-ui/pull/1775

The test fails in this PR for the environment variable because the test uses `main`, which doesn't have this change yet.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -59,4 +59,4 @@ To preview the changes you made for this Amplify UI Docs, you can use Amplify to
 Then you need to:
 
 1. After selecting a branch, check "Connecting a monorepo? Pick a folder." and at "Enter the root directory of your app (e.g. src/myapp)", enter "docs"
-2. Set "Environment variables" from the left sidebar. Add "SITE_URL" and set it to a string value.
+2. Set "Environment variables" from the left sidebar. Add "SITE_URL", "DOCSEARCH_DOCS_APP_ID", "DOCSEARCH_DOCS_API_KEY", "DOCSEARCH_DOCS_INDEX_NAME".
```
[the remark this answers is not in the document]
Comment (0618) [reply docs/README.md:62]: Good point...Should we just leave the keys README or `.env.example`?

[the remark this answers is not in the document]
Comment (0618) [reply docs/README.md:62]: Thanks, Scott! I added testing credentials 😁

File: docs/README.md [hunk 2]
```diff
@@ -59,4 +59,11 @@ To preview the changes you made for this Amplify UI Docs, you can use Amplify to
...
+   - "SITE_URL": feel free to use any string
+   - "DOCSEARCH_DOCS_APP_ID", "DOCSEARCH_DOCS_API_KEY", "DOCSEARCH_DOCS_INDEX_NAME". If you have them from [Algolia DocSearch](https://docsearch.algolia.com/), use them. Otherwise, please use DocSearch's [testing credentials](https://docsearch.algolia.com/docs/DocSearch-v3)
```
Comment (slaymance) [inline docs/README.md:64]: The Algolia DocSearch docs linked here aren't very clear on how to use the App ID and API Key. Can we clarify this part a bit more?

Comment (0618) [reply docs/README.md:64]: What would you suggest to be be added besides in addition to L62?

Comment (slaymance) [reply docs/README.md:64]: Well I don't really have a suggestion, the line "_If you have them from Algolia DocSearch_" just isn't clear to me.

Comment (0618) [reply docs/README.md:64]: That should be fine. That means you don't "have them from Algolia DocSearch" 😁

**Review by reesscot — APPROVED** [review]

**Comment by 0618** [comment]
Hold off for https://github.com/aws-amplify/amplify-ui/pull/1955

File: .github/workflows/reusable-e2e.yml [hunk 3]
```diff
@@ -307,6 +307,9 @@ jobs:
         run: yarn docs build
         env:
           SITE_URL: ${{ secrets.SITE_URL }}
+          DOCSEARCH_DOCS_APP_ID: ${{ secrets.DOCSEARCH_DOCS_APP_ID }}
+          DOCSEARCH_DOCS_API_KEY: ${{ secrets.DOCSEARCH_DOCS_API_KEY }}
+          DOCSEARCH_DOCS_INDEX_NAME: ${{ secrets.DOCSEARCH_DOCS_INDEX_NAME }}
```
Comment (wlee221) [inline .github/workflows/reusable-e2e.yml:312]: Can you add these three to the secret list at the top of this file?

https://github.com/aws-amplify/amplify-ui/blob/2cd71b992bfec493c7b7c274ef6f86c63710cf7e/.github/workflows/reusable-e2e.yml#L13-L28

Comment (0618) [reply .github/workflows/reusable-e2e.yml:318]: Thanks! Done!

Comment (0618) [reply .github/workflows/reusable-e2e.yml:318]: Thanks! Done!

**Review by wlee221 — APPROVED** [review]
Thanks @0618!

**Pull request merged**